pegen/c_generator.py

```diff
@@ -276,6 +276,14 @@ def _handle_default_rule_body(self, node: Rule, rhs: Rhs, result_type: str) -> N
                 with self.indent():
                     self.print("return res;")
             self.print("int mark = p->mark;")
+            self.print("if (p->mark == p->fill && fill_token(p) < 0) {")
```

Comment:
Suggestion: Could we factor this into a method? This way we can easily identify
in the future what this new chunk is for and also it seems very similar (if not the same)
with the new addition in `_handle_loop_rule_body` and part of `visit_Alt`.

Reply:
What about two methods? One for lineno/col_offset and one for end_lineno/end_col_offset?

Reply:
That would be fine, although I think I prefer only as they are still very related. We can add maybe some comments before the lineno/col_offset  and the end_lineno/end_col_offset part. Being said that, I don't feel strongly either way :)

Reply:
Would what I've done in 7a4f0f4 work?

Reply:
> Would what I've done in [7a4f0f4](https://github.com/gvanrossum/pegen/commit/7a4f0f4d28246e70182d1e6460e45570327e0637) work?

Yeah, that is fantastic. Thanks!